algos.js

```diff
@@ -74,6 +74,39 @@
         }
         return list;
     };
+    var randomizedPartition, randomizedQuickSort;
+    randomizedPartition = function(list, left, right) {
+        var pivot = list[Math.floor(Math.random() * (right - left) + left)];
+        while (left <= right) {
+            while (list[left] < pivot) {
+                left += 1;
+            }
+            while (list[right] > pivot) {
+                right -= 1;
+            }
+            if (left <= right) {
+                swap(list, left, right);
+                left += 1;
+                right -= 1;
+            }
+        }
+        return left;
+    };
```

Comment:
This is an example feedback:

You could add a single whitespace line here for logic separation.